backends/graphs/graphs.h

```diff
@@ -14,8 +14,8 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-#ifndef _BACKENDS_GRAPHS_CONTROLS_H_
```

Comment:
it's a little confusing to call this file `graphs.h`, given that none of the classes it prototypes are implemented in `graphs.cpp`.

Reply:
It is the common header file for both control.cpp and parser.cpp.
If you have any other suggestion for names, i would appreciate that.

Reply:
Maybe `graph_generators.h` if you want to have a single header file for both control and parser

Reply:
If you refactor `ParserGraphs` and `ControlGraphs` into a single `Graphs` (abstract) base class with `ParserGraphs` and `ControlGraphs` as subclasses, then I think this organization does make a fair amount of sense.

Reply:
I tried to refactor the classes and created a Graphs base class.